File: daemon/Models/Database/Journey/Journey.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace daemon.Models.Database.Journey;

public class Journey
{
    [Key]
    [MaxLength(128)]
    [Column("journey_id")]
    public required string Id { get; init; }

    [Column("date")]
    public required DateOnly Date { get; init; }

    [Column("inserted_at")] 
    public required DateTime InsertedAt { get; init; }

    [Column("type")] 
    public required JourneyType Type { get; init; } = JourneyType.REGULAR;
    
    
    [Column("administration_index")]
    public int AdministrationIndex { get; init; }

    [ForeignKey(nameof(AdministrationIndex))] 
    public required Administration Administration { get; init; }
    
    
    [Column("cancelled")]
    public required bool Cancelled { get; init; } = false;
    
    
    public virtual Transport? Transport { get; init; }
    
    public virtual ICollection<ScheduleAtStopPlace> ViaStops { get; init; } = new List<ScheduleAtStopPlace>();
}

File: backend/Sources/backend/Models/Journey/Journey.swift
//
//  Journey.swift
//  backend
//
//  Created by Christian Knapp on 21.09.25.
//

import Fluent
import Foundation

final class Journey: Model, @unchecked Sendable {
    static let schema: String = "journeys"
    static let space: String? = "core"
    
    @ID(custom: "journey_id", generatedBy: .user)
    var id: String?
    
    @Field(key: "date")
    var date: Date
    
    @Field(key: "inserted_at")
    var insertedAt: Date
    
    @Field(key: "type")
    var type: JourneyType
    
    @Parent(key: "administration_index")
    var administration: Administration
    
    @Field(key: "cancelled")
    var cancelled: Bool
    
    @OptionalChild(for: \.$journey)
    var transport: Transport?
    
    @Children(for: \.$journey)
    var viaStops: [ScheduleAtStopPlace]
    
    init() { }
    
    init(journeyId: String, date: Date, type: JourneyType, administrationIndex: Administration.IDValue) {
        self.id = journeyId
        self.date = date
        self.type = type
        self.$administration.id = administrationIndex
    }
}
